Library/Library.cs

```diff
@@ -29,18 +33,76 @@ private void Initialize()
             LibraryDictionary.Add("Sita", 2);
             LibraryDictionary.Add("Increase Productivity", 4);
             LibraryDictionary.Add("C sharp", 5);
-            LibraryDictionary.Add("Head First Java", 1);
+            LibraryDictionary.Add("JavaBook", 1);// test with book
         }
 
-        public bool Issue(string bookName)
+        public bool Issue(string bookName, string userName) // add username parameter
         {
-            if (LibraryDictionary.ContainsKey(bookName) && LibraryDictionary[bookName] > 0)
+            if (LibraryDictionary.ContainsKey(bookName))
             {
-                LibraryDictionary[bookName] = LibraryDictionary[bookName] - 1 ;
-                return true;
+                if (LibraryDictionary[bookName] > 0)
+                {
+                    LibraryDictionary[bookName] = LibraryDictionary[bookName] - 1;
+                    // add entry to dictionary issue dictioary
+                    IssuedBooks.Add(bookName, userName);
+                    return true;
+                }
+                else
+                // else block that means book count is 0
+                // add this bookname and user name mapping to hold dictionary
+                HoldBooks.Add(bookName, userName);
             }
             return false;
         }
+
+        public bool Return(string bookName, string userName) // add paramter username
+        {
+            if (LibraryDictionary.ContainsKey(bookName))
+            {
+                // remove user1 person entry from issue dictionary
+                IssuedBooks.Remove(bookName);
+                // check if hold dictionary contains entry for thod book (user2 is in hold dictionary)
+                if (HoldBooks.ContainsKey(bookName))
+                {
+                   string Holdperson = HoldBooks[bookName];
```

Comment:
local variable should start with small char : holdPerson. search in all classes and make the similar change